client_new/src/tests/network_tests.rs

```diff
@@ -0,0 +1,815 @@
+    let remote_client_path = rand::bytestring(1024);
+    let remote = Stronghold::default();
+    let config = NetworkConfig::new(Permissions::allow_all()).with_mdns_enabled(true);
+
+    // we need to create a client on the remote that accepts incoming requests
+    let result = remote.create_client(remote_client_path.clone());
+    assert!(result.is_ok(), "Assertion Failed= {:?}", result);
+
+    let client = result.unwrap();
+    let result = client.write_p2p_keypair(remote_key_pair, remote_key_path.clone());
+    assert!(result.is_ok(), "Assertion Failed=  {:?}", result);
+
+    let result = remote
+        .spawn_p2p(remote_client_path.clone(), config, Some(remote_key_path))
+        .await;
+    assert!(result.is_ok(), "Assertion Failed=  {:?}", result);
+
+    let (mut sender_terminate_signal, receiver_terminate_signal) = futures::channel::mpsc::unbounded();
+
+    let result = remote.start_listening(None).await;
+    assert!(result.is_ok(), "Assertion Failed= {:?}", result);
+
+    let remote_address = result.unwrap();
+
+    let remote_peer_id = PeerId::from_public_key(&remote_public_key);
+
+    // clone remote which will be moved into a background task
+    let remote_stronghold_server = remote.clone();
+
+    // keep handle to server
+    let server = tokio::spawn(async move { remote_stronghold_server.serve(receiver_terminate_signal).await });
+    // tests come here
+    {
+        let local_client_path = rand::bytestring(1024);
+        let local_key_pair = Keypair::generate_ed25519();
+        let local_public_key = local_key_pair.public();
+        let local_key_path = Location::generic(b"keypair-path".to_vec(), b"keypair-path".to_vec());
+        let config = NetworkConfig::new(Permissions::allow_all()).with_mdns_enabled(true);
+        let local = Stronghold::default();
+
+        let result = local.create_client_with_keys(local_client_path.clone(), local_key_pair, local_key_path);
+        assert!(result.is_ok(), "Assertion Failed= {:?}", result);
+
+        let result = local.spawn_p2p(local_client_path, config, None).await;
+        assert!(result.is_ok(), "Assertion Failed= {:?}", result);
+
+        let result = local.add_peer_addr(remote_peer_id, remote_address).await;
+        assert!(result.is_ok(), "Assertion Failed= {:?}", result);
+
+        let local_address = result.unwrap();
+        let result = remote
+            .add_peer_addr(PeerId::from_public_key(&local_public_key), local_address)
+            .await;
+        assert!(result.is_ok(), "Assertion Failed= {:?}", result);
+
+        // create peer
+        let result = local.create_remote_client(remote_public_key, remote_client_path).await;
+        assert!(result.is_ok(), "Assertion Failed= {:?}", result);
+
+        // connect peer
+        let peer = result.unwrap();
+        let result = peer.connect().await;
+        assert!(result.is_ok(), "Assertion Failed= {:?}", result);
```

Comment:
Why can't `connect` be a private method and be called from within `create_remote_client`? This seems like an unnecessary extra step. What if I call `peer.remote_write_store` before calling `peer.connect`? I assume I get an error? 
Would be nice if obtaining an instance of a `Peer` would mean that I'm in a connected state right from the start.

Reply:
The idea was to have two separate atomic operations, one to obtain a saved peer, the other to connect. Only a `disconnect` function is missing. I can evaluate if a `Peer` can be connected right away